Source/WebCore/inspector/NetworkResourcesData.cpp

```diff
@@ -260,6 +260,13 @@ void NetworkResourcesData::maybeDecodeDataToContent(const String& requestId)
     size_t dataLength = resourceData->content().sizeInBytes();
     if (dataLength > m_maximumSingleResourceContentSize)
         m_contentSize -= resourceData->evictContent();
+
+    while (m_contentSize > m_maximumResourcesContentSize) {
+        String requestId = m_requestIdsDeque.takeFirst();
```

Comment:
hmm im not really sure about this.  do we know for sure what the ordering of resources is inside of this?  i have a feeling that the first thing added is the `.html`, which we probably never want to evict

i think our current existing behavior is that if the developer requests enough things such that we run out of space, Web Inspector will show an error when trying to preview the resource content

i can definitely understand with and agree that that's not a great experience, but i think we need to be smarter about what things to evict from the cache

Reply:
This approach was taken by me base how it is done in `NetworkResourcesData::ensureFreeSpace`. If it is wrong we should fix that also there (separate PR?).

Some idea for another PR - do we need to gather data which is not able to be displayed in inspector (video or part of it)?

Reply:
hmmm maybe it's actually fine to get rid of the oldest resource since it should've already been sent to the frontend?

Reply:
hmm, actually if we have a site which loads a lot of resources and it exceeds the limit of total resources cache, than webkit removes the main document resource data from the cache and it will not be available in web inspector.